src/state/profile-types/actions.js

```diff
@@ -1,18 +1,300 @@
-import { SET_PROFILE_TYPES } from 'state/profile-types/types';
-import { getProfileTypes } from 'api/profileTypes';
+
+
+import { gotoRoute, getParams } from '@entando/router';
 import { setPage } from 'state/pagination/actions';
+import { toggleLoading } from 'state/loading/actions';
+import { addErrors } from 'state/errors/actions';
+import { initialize } from 'redux-form';
+
+import {
+  ROUTE_PROFILE_TYPE_LIST,
+  ROUTE_PROFILE_TYPE_EDIT,
+  ROUTE_ATTRIBUTE_MONOLIST_ADD,
+} from 'app-init/router';
 
+import {
+  postProfileType,
+  putProfileType,
+  deleteProfileType,
+  getProfileType,
+  getProfileTypes,
+  deleteAttributeFromProfileType,
+  getAttributeFromProfileType,
+  postAttributeFromProfileType,
+  putAttributeFromProfileType,
+  getProfileTypeAttributes,
+  getProfileTypeAttribute,
+} from 'api/profileTypes';
+import {
+  SET_PROFILE_TYPES,
+  REMOVE_PROFILE_TYPE,
+  REMOVE_ATTRIBUTE,
+  SET_ATTRIBUTES,
+  SET_SELECTED_PROFILE_TYPE,
+  SET_SELECTED_ATTRIBUTE_FOR_PROFILETYPE,
+  SET_SELECTED_ATTRIBUTE,
+}
+  from 'state/profile-types/types';
+import { getProfileTypeAttributesIdList, getProfileTypeSelectedAttributeType } from 'state/profile-types/selectors';
 
+const TYPE_MONOLIST = 'Monolist';
+
+// Data type
 export const setProfileTypes = profileTypes => ({
   type: SET_PROFILE_TYPES,
   payload: {
     profileTypes,
   },
 });
 
+export const removeProfileType = profileTypeCode => ({
+  type: REMOVE_PROFILE_TYPE,
+  payload: {
+    profileTypeCode,
+  },
+});
+
+export const setSelectedProfileType = profileType => ({
+  type: SET_SELECTED_PROFILE_TYPE,
+  payload: {
+    profileType,
+  },
+});
+
+export const setSelectedAttributeProfileType = attribute => ({
+  type: SET_SELECTED_ATTRIBUTE_FOR_PROFILETYPE,
+  payload: {
+    attribute,
+  },
+});
+
+export const removeAttribute = (profileTypeCode, attributeCode) => ({
+  type: REMOVE_ATTRIBUTE,
+  payload: {
+    profileTypeCode,
+    attributeCode,
+  },
+});
+
+// Data type attributes
+export const setSelectedAttribute = attribute => ({
+  type: SET_SELECTED_ATTRIBUTE,
+  payload: {
+    attribute,
+  },
+});
+
+export const setProfileTypeAttributes = attributes => ({
+  type: SET_ATTRIBUTES,
+  payload: {
+    attributes,
+  },
+});
+
+
 // thunk
-export const fetchProfileTypes = (page = 1, params) => dispatch =>
-  getProfileTypes(page, params).then((data) => {
-    dispatch(setProfileTypes(data.payload));
-    dispatch(setPage(data.metaData));
+
+export const sendPostProfileType = ProfileTypeObject => dispatch =>
+  new Promise((resolve) => {
+    postProfileType(ProfileTypeObject).then((response) => {
+      response.json().then((json) => {
```

Comment:
What if you get error 503 or similar (with no JSON content)? You should make sure the response has a JSON format before reading. you can read the headers, or call response.json() in a try/catch block, handling the error with a Promise rejection for example